### PR TITLE
LANDGRIF-373 - Refactor h3Data - material relationship

### PR DESCRIPTION
**Pull request by tiagojsag**

### CODE REVIEW

File: api/src/modules/h3-data/dto/get-material-h3-by-resolution.dto.ts [hunk 1]
```diff
@@ -27,4 +27,10 @@ export class GetMaterialH3ByResolutionDto {
   @IsNumber()
   @IsEnum(AvailableResolutions, { message: 'Available resolutions: 1 to 6' })
   resolution!: number;
+
+  @ApiProperty()
+  @Type(() => Number)
+  @IsNumber()
+  @IsNotEmpty()
+  year!: number;
 }
```
Comment (alexeh) [inline api/src/modules/h3-data/dto/get-material-h3-by-resolution.dto.ts:36]: Right, this param was needed indeed and I should have added it myself. thanks!

File: api/src/modules/materials/materials.service.ts [hunk 2]
```diff
@@ -60,7 +60,7 @@ export class MaterialsService extends AppBaseService<
    */
   static filterMaterialTree(root: Material): Material[] {
     let result: Material[] = [];
-    if (root.producerId === null && root.harvestId === null) {
+    if (!root.materialToH3s || root.materialToH3s.length === 0) {
```
Comment (alexeh) [inline api/src/modules/materials/materials.service.ts:63]: I think this could also be shorthanded to  !root.materialToH3s.length

**Review by yulia-bel — APPROVED** [review]

**Comment by tiagojsag** [comment]
Thanks for the feedback, I think I got all of your comments addressed, please let me know if I missed anything

**Review by alexeh — APPROVED** [review]
LGTM, Thanks for this BIG task!

**Pull request merged**